migrations/versions/20181220_populate_measures.py

```diff
@@ -0,0 +1,164 @@
+"""Populate topic, subtopic, subtopic_measure and measure table from the main page table.
+
+Revision ID: 20181220_populate_measures
+Revises: 20181220_measureid
+Create Date: 2018-12-11 13:11:47.567135
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = "20181220_populate_measures"
+down_revision = "20181220_measureid"
+branch_labels = None
+depends_on = None
+
+max_measure_version_temp_table = """
+CREATE TEMPORARY TABLE max_measure_version
+AS
+SELECT
+    mv.*
+FROM
+    measure_version AS mv
+WHERE
+    mv.page_type = 'measure'
+    AND mv.version = (
+        SELECT
+            MAX(version)
+        FROM
+            measure_version AS max_mv
+        WHERE
+            max_mv.guid = mv.guid);
+"""
+
+copy_topics = """
+INSERT INTO topic (uri, title, description, additional_description)
+SELECT DISTINCT
+    uri, title, description, additional_description
+FROM
+    measure_version
+WHERE
+    page_type = 'topic'
+"""
+
+copy_subtopics = """
+INSERT INTO subtopic (uri, title, position, topic_id)
+SELECT DISTINCT
+    mv_subtopic.uri, mv_subtopic.title, mv_subtopic.position, topic.id
+FROM
+    measure_version AS mv_subtopic
+JOIN
+    measure_version AS mv_topic
+ON
+    mv_subtopic.parent_guid = mv_topic.guid
+    AND mv_subtopic.parent_version = mv_topic.version
+JOIN
+    topic
+ON
+    mv_topic.uri = topic.uri
+    AND mv_topic.description = topic.description
+    AND mv_topic.additional_description = topic.additional_description
+WHERE
+    mv_subtopic.page_type = 'subtopic'
+"""
+
+copy_measures = """
+INSERT INTO measure (guid, uri, position, reference)
+SELECT
+    guid,
+    uri,
+    position,
+    internal_reference
+FROM
+    max_measure_version;
+"""
+
+create_subtopic_measure_join_records = """
+INSERT INTO subtopic_measure (subtopic_id, measure_id)
+SELECT DISTINCT
+    subtopic.id, measure.id
+FROM
+    measure_version AS mv_subtopic
+JOIN
+    measure_version AS mv_measure
+ON
+    mv_subtopic.id = mv_measure.parent_id
+JOIN
+    subtopic
+ON
+    subtopic.uri = mv_subtopic.uri
+    AND mv_subtopic.page_type = 'subtopic'
+JOIN
+    measure
+ON
+    measure.uri = mv_measure.uri
```

Comment:
I think this might be where the problem creeps in - because measure URIs aren't unique. So adding `AND measure.guid = mv_measure.guid` might fix it.

Reply:
Done, works. 👍